scripts/tf-tests/README.md

```diff
@@ -0,0 +1,117 @@
+# Manual integration tests using Testflinger
+
+The purpose of these scripts is to test hardware-api client and server
+on the machines that are accessible via
+[Testflinger](https://github.com/canonical/testflinger). It allows us
+to test the project components on multiple machines automatically.
+
+## Requirements
+
+These scripts use [the testflinger
+snap](https://snapcraft.io/testflinger-cli), so make sure you have it
+installed on your system:
+
+```sh
+sudo snap install testflinger-cli
+```
+
+Also, the following files are required to be present in this directory:
+
+- `machines.txt`: This file lists the Canonical IDs of the machines on
+  which jobs will be run. Each Canonical ID should be on a separate
+  line, formatted as follows:
+
+  ```text
+  202101-28595
+  202012-28526
+  ...
+  ```
+
+- `tf-job.yaml`: This YAML template defines the job parameters. The
+  template should include the placeholder `$CANONICAL_ID`, which will
+  be replaced with each actual Canonical ID from `machines.txt` when
+  jobs are submitted.  You can modify the existing `tf-job.yaml` file
+  to run other commands or use a different Ubuntu distro.
+
+- `hwctl`: The client executable. You can create it by running `cargo
+  build --release` in the project root directory. You're supposed to
+  build the binary on the same Ubuntu release that is specified in
+  `tf-job.yaml`. The machines should also be of the same architecture
+  the binary is build for.
+
+  Then copy the created file to this directory:
+
+  ```sh
+  cp  target/release/hwctl scripts/tf-tests/
+  ```
+
+## Running the scripts
+
+After you meel the described requirements, make sure you have access
+to
+[https://testflinger.canonical.com](https://testflinger.canonical.com).
+
+These scripts are designed to work sequentially, where `run-jobs.py`
+must be executed first to submit jobs and create job directories,
+followed by `check-status.py` to monitor and fetch job results.
```

Comment:
Is there any chance that someone might execute `run-jobs.py` and then _not_ execute `check-status.py`? If not, wouldn't it make sense if these two scripts were merged into one?

Reply:
Yeah, I did it for myself because sometimes I don't want to track job outputs immediately but do it later. Also, the `check-status` script allows us to save jobs output without running new jobs.

Reply:
I've combined these scripts into one